Global changes (#1526)
hadley/adv-r: Foundations.Rmd

To start your journey in mastering R, the following six chapters will help you learn the foundational components of R. I expect that you've already seen many of these pieces before, but you probably have not studied them deeply. To help check your existing knowledge, each chapter starts with a quiz; if you get all the questions right, feel free to skip to the next chapter!

> 1.  Chapter \@ref(names-values), teaches you about an important distinction

Reviewer: ```suggestion
1.  Chapter \@ref(names-values) teaches you about an important distinction
```

    that you probably haven't thought deeply about: the difference between an 
    object and its name. Improving your mental model here will help you make 
    better predictions about when R copies data and hence which basic 
    operations are cheap and which are expensive.
   
1.  Chapter \@ref(vectors-chap) dives into the details of vectors, helping you 
    learn how the different types of vector fit 
    together. You'll also learn about attributes, which allow you to store 
    arbitrary metadata, and form the basis for two of R's object-oriented 
    programming toolkits.
    

> 1.  Chapter \@ref(subsetting) describes how to use subsetting properly to write 

Reviewer: ```suggestion
1.  Chapter \@ref(subsetting) describes how to use subsetting to write 
```

    clear, concise, and efficient R code. Understanding the fundamental 
    components will allow you to solve new problems by combining the building 
    blocks in novel ways.

1.  Chapter \@ref(control-flow) presents tools of control flow that allow you to
    only execute code under certain conditions, or to repeatedly execute code 

>     with changing inputs: the important `if` and `for` constructs, as well as 

Reviewer: ```suggestion
    with changing inputs. These include important `if` and `for` constructs, as well as 
```

>     with changing inputs. These include important `if` and `for` constructs, as well as 

Reviewer: ```suggestion
    with changing inputs. These include the important `if` and `for` constructs, as well as 
```

    related tools like `switch()` and `while`.

> 1.  Chapter \@ref(functions) deals with functions -- the most important builing 

Reviewer: ```suggestion
1.  Chapter \@ref(functions) deals with functions, the most important building 
```
